tests/Bech32Test.php

```diff
@@ -0,0 +1,297 @@
+<?php
+
+namespace CardanoPhp\Bech32;
+
+use Exception;
+use PHPUnit\Framework\Attributes\CoversClass;
+use PHPUnit\Framework\TestCase;
+
+#[CoversClass(Bech32::class)]
+class Bech32Test extends TestCase
```

Comment:
Minor: 

* Add Tests for Empty Data Arrays:
  * Ensure that encoding with empty data arrays produces the expected Bech32 string format.

* Enhance HRP Validation:
  * Implement and test checks to ensure that the HRP contains only valid Bech32 characters.

Reply:
This has been created as a new issue and marked as a "Good First Issue" for a contributor